src/lib/components/wizard.component.ts

```diff
@@ -107,24 +107,81 @@ export class WizardComponent implements OnChanges, AfterContentInit {
 
   /**
    * The initially selected step, represented by its index
+   * Beware: This initial default is only used if no wizard step has been enhanced with the `selected` directive
    */
   @Input()
-  public defaultStepIndex = 0;
+  public get defaultStepIndex(): number {
+    // This value can be either:
+    // - the index of a wizard step with a `selected` directive, or
+    // - the default step index, set in the [[WizardComponent]]
+
+    const foundDefaultStep = this.wizardSteps.find(step => step.defaultSelected);
+
+    if (foundDefaultStep) {
+      return this.getIndexOfStep(foundDefaultStep);
+    } else {
+      return this._defaultStepIndex;
+    }
+  }
+  public set defaultStepIndex(defaultStepIndex: number) {
+    this._defaultStepIndex = defaultStepIndex;
+  }
+  private _defaultStepIndex = 0;
+
+  /**
+   * The WizardStep object belonging to the currently visible and selected step.
+   * The currentStep is always the currently selected wizard step.
+   * The currentStep can be either completed, if it was visited earlier,
+   * or not completed, if it is visited for the first time or its state is currently out of date.
+   *
+   * If this wizard contains no steps, currentStep is null
+   */
+  public get currentStep(): WizardStep {
+    if (this.hasStep(this.currentStepIndex)) {
+      return this.wizardSteps[this.currentStepIndex];
+    } else {
+      return null;
+    }
+  }
 
   /**
    * True, if the navigation bar shouldn't be used for navigating
    */
   @Input()
   public disableNavigationBar = false;
 
+  /**
+   * An array representation of all wizard steps belonging to this model
+   */
+  public get wizardSteps(): WizardStep[] {
+    return this._wizardSteps;
+  }
+  private _wizardSteps: WizardStep[] = [];
```

Comment:
I think Codacy will complain about the ordering of the fields inside the class (mixing of public and private fields)

Reply:
Do you think we should move properties and their backing fields apart? I think it will harm readability.

Reply:
I believe this depends on what you expect. In other languages (Java for example) you typically divide private/public/protected fields and methods from each other by adding them in a certain order to your class (for me private members first, public members last).

This is also the case for fields vs methods/functions, where the fields should be added first followed by the methods/functions. For me properties (i.e. getter and setter methods) are also methods in their core, which can be accessed in a special way. Therefore I think they should also be added to the "method section" in the code.

Reply:
Thanks for explaining your reasoning! Done in bd58cb12.

I am not sure whether we need to do something with the `defaultStepIndex` property. It is also public, but on the other hand it is an `@Input`, and generally all `@Input`'s are placed in the beginning of the class before the constructor. For now I have left it intact.

Reply:
In this case I think it's ok to leave the `defaultStepIndex` property where it is currently. 
It is more about the overall layout of the fields/methods inside the classes.